src/main/java/io/neonbee/internal/verticle/ServerVerticle.java

```diff
@@ -1,366 +1,283 @@
-        addEndpointHandler(router, DEFAULT_METRICS_BASE_PATH, EMPTY,
-                endpointsConfig.getJsonObject("metrics", new JsonObject()), (basePath,
-                        endpointConfig) -> PrometheusScrapingHandler.create(endpointConfig.getString("registryName")));
-        router.route().handler(NotFoundHandler.create());
-
-        NeonBeeOptions options = NeonBee.instance(vertx).getOptions();
-        int port = Optional.ofNullable(options.getServerVerticlePort())
-                .orElse(config.getInteger(CONFIG_PROPERTY_PORT_KEY, DEFAULT_PORT));
-
-        httpServer = vertx.createHttpServer(new HttpServerOptions()
-                .setDecompressionSupported(config.getBoolean("decompressionSupported", true))
-                .setCompressionSupported(config.getBoolean("compressionSupported", true))
-                .setCompressionLevel(config.getInteger("compressionLevel", 1))
-                .setMaxHeaderSize(config.getInteger("maxHeaderSize", DEFAULT_MAX_HEADER_SIZE))
-                .setMaxInitialLineLength(config.getInteger("maxInitialLineLength", DEFAULT_MAX_INITIAL_LINE_LENGTH))
-                .setUseAlpn(config.getBoolean("useAlpn", true))).exceptionHandler(throwable -> {
-                    LOGGER.error("HTTP Socket Exception", throwable);
-                }).requestHandler(router).listen(port, asyncResult -> {
-                    if (asyncResult.succeeded()) {
-                        LOGGER.info("HTTP server started on port {}", port);
-                        if (LOGGER.isDebugEnabled()) {
-                            LOGGER.debug("HTTP server configured with routes: {}",
-                                    router.getRoutes().stream().map(Route::toString).collect(Collectors.joining(",")));
-                        }
-                        startPromise.complete();
-                    } else {
-                        LOGGER.error("HTTP server could not be started on port {}", port, asyncResult.cause()); // NOPMD
-                        startPromise.fail(asyncResult.cause());
-                    }
+        // create the default authentication (chain) and hooks handlers
+        Optional<AuthenticationHandler> defaultAuthHandler = createAuthHandler(config.getAuthChainConfig());
+        HooksHandler hooksHandler = HooksHandler.create();
+
+        // add all endpoint handlers as sub-routes here
+        mountEndpoints(router, config.getEndpointConfigs(), defaultAuthHandler, hooksHandler)
+                .onFailure(startPromise::fail).onSuccess(nothing -> {
+                    // register a fallback route and the NotFoundHandler, which finally fails
+                    Route fallbackRoute = router.route();
+                    defaultAuthHandler.ifPresent(fallbackRoute::handler);
+                    fallbackRoute.handler(hooksHandler).handler(NotFoundHandler.create());
+
+                    vertx.createHttpServer(config /* ServerConfig is a HttpServerOptions subclass */)
```

Comment:
Is this the only reason why ServerConfig is extending HttpServerOptions? If yes we could maybe introduce a "toHttpServerOptions()" method. Then we would also solve the issue, that we have a lot of attributes in ServerConfig  which are not stored in the json.

Reply:
I was wrong here, they are stored